osu.Game/Overlays/ModButton.cs

```diff
@@ -0,0 +1,100 @@
+﻿//Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
+//Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK.Graphics;
+using osu.Framework;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input;
+using osu.Game.Graphics;
+using osu.Game.Modes;
+using osu.Framework.Extensions;
+
+namespace osu.Game.Overlays
+{
+    class ModButton : ClickableContainer, IStateful<ModButtonState>
+    {
+        private TextAwesome icon, bg;
+        private Color4 bgColor;
+
+        public ModButtonState State{get; set;}
+        private Mod Mod;
+
+        private const int transform_time = 150;
+
+        public ModButton(Mod mod, Color4 color, bool isActive)
+        {
+            Mod = mod;
+            bgColor = color;
+
+            State = ModButtonState.Disabled;
+
+            RelativeSizeAxes = Axes.Y;
+            Width = 100;
+
+            Origin = Anchor.Centre;
+            Anchor = Anchor.CentreLeft;

```

Comment:
i'd extract the display logic out from the button logic. we'll want to display mod icons in other places in the same way here but not have them interactive.

`ModButton : DrawableMod` or something.